Core api reference refresh (#669)
ROCm/rocDecode: docs/how-to/using-rocDecode-rocdecdecoder.rst

.. meta::
  :description: Using the rocDecode core API
  :keywords: rocDecode, AMD, ROCm, core API

********************************************************************
Using the rocdecdecode example
********************************************************************

rocDecode provides four core APIs exposed in the header files in the |apifolder|_ directory:

> 

Reviewer: These should be bullets:
<img width="1151" height="360" alt="image" src="https://github.com/user-attachments/assets/aa5ca22e-329f-4446-9aae-50212cdc4a99" />

Author: This isn't really a list. It's more that I split these up for readability.

| The rocDecode parser API, exposed in ``rocparser.h``. 
| The hardware decoder API, exposed in ``rocdecode.h``.
| The software decoder API, exposed in ``rocdecode_host.h``.
| The bitstream reader API, exposed in ``roc_bitstream_reader.h``.

The |rocdecdecode|_ sample demonstrates how to use the rocDecode core APIs in an application. It shows how to use the parser and both the hardware and software decoders. For information on how to use the bitstream reader API, see :doc:`Using the rocDecode bitstream reader API <./using-rocDecode-bitstream>`.

> The sample decodes raw elementary video frame files as input and produces individually decoded frames in YUV format as output. The input can be one individual frame file or multiple frames from one or more video files. The individual frame files must be numbered in ascending order of frames.

Reviewer: ```suggestion
The sample decodes raw elementary video frame files as input and produces individually decoded frames in YUV format as output. The input can be one individual frame file or multiple frames from one or more video files. The individual frame files must be numbered in ascending order, frame number.

```

Author: The files need to be numbered in ascending order of frames.

``rocdecdecode.cpp`` takes the following arguments:

.. list-table:: 
    :widths: 10 60 30 
    :header-rows: 1

> 

Reviewer: Should these be alpha order?
Perhaps the descriptions with multiple options should be bullets to make it easier to scan.

Reviewer: <img width="1188" height="908" alt="image" src="https://github.com/user-attachments/assets/4fe332d0-4e02-4c29-81c5-1c3a052720e7" />

Author: These are how they're listed in the readme. On the one hand, I see the value of alphabetical order, but on the other, I find that having them reflect the order in the readme to be more useful so you don't have to go searching.

Author: Also the lack of bullet points is the same as in other docs in rocDecode

    * - ``-d`` 
      - GPU device ID. Set it to 0 for the first device, 1 for the second device, 2 for the third device, and so on for each subsequent device. 
      - Optional. Set to 0 by default.
  
    * - ``-b`` 
      - Backend. Set it to 0 to use the hardware decoder on the GPU or to 1 to use the software decoder on the CPU.
      - Optional. Set to 0 by default.

    * - ``-c``
      - Codec. Set to 0 for HEVC, 1 for H264, 2 for AV1, 4 for VP9, 5 for VP8, or 6 for MJPEG.
      - Optional. Set to 0 by default.
      
    * - ``-n`` 
      - Number of iterations for performance evaluation.
      - Optional. Set to 1 by default.

    * - ``-m`` 
      - The output surface memory type. The memory type where the surface data, such as the decoded frames, resides. Set this to 0 for intermediate GPU memory, to 1 for GPU memory, and to 2 for CPU memory. See :doc:`Surface data memory locations <../conceptual/rocDecode-memory-types>` for more information. 
      - Optional. Set to 0 by default. 

The ``DecoderInfo`` struct defined in the sample is used to store user-supplied parameters as well as the decoder and parser handles. 

> The memory type and the type of decoder is set by the specified backend. If the GPU (device) backend is selected, a parser is created along with a decoder. If the CPU (host) backend is selected, then only a software decoder is created:

Reviewer: ```suggestion
The memory type and the type of decoder is set by the specified backend. If the GPU (device) backend is selected, a parser and a decoder are created. If the CPU (host) backend is selected, then only a software decoder is created:

```
